examples/platforms/posix/ble_nimble.c

```diff
@@ -0,0 +1,1111 @@
+// otError otPlatBleGattVendorUuidRegister(otInstance *aInstance, const otPlatBleUuid *aUuid);
+
+otError otPlatBleGattMtuGet(otInstance *aInstance, uint16_t *aMtu)
+{
+    (void)aInstance;
+    (void)aMtu;
+
+    return OT_ERROR_NOT_IMPLEMENTED;
+}
+
+static int gatt_event_cb(uint16_t conn_handle, uint16_t attr_handle, struct ble_gatt_access_ctxt *ctxt, void *arg)
+{
+    int         rc       = 0;
+    otInstance *instance = (otInstance *)arg;
+
+    (void)conn_handle;
+
+    switch (ctxt->op)
+    {
+    case BLE_GATT_ACCESS_OP_WRITE_CHR:
+    {
+        otBleRadioPacket packet;
+        packet.mValue  = OS_MBUF_DATA(ctxt->om, uint8_t *);
+        packet.mLength = OS_MBUF_PKTLEN(ctxt->om);
+
+        dispatch_otPlatBleGattServerOnWriteRequest(instance, attr_handle, &packet);
+        break;
+    }
+
+    case BLE_GATT_ACCESS_OP_READ_CHR:
+    {
+        // NOTE: Not needed by WoBLE, as indicate is sent directly.
+        otBleRadioPacket packet;
+        dispatch_otPlatBleGattServerOnReadRequest(instance, attr_handle, &packet);
+
+        rc = os_mbuf_append(ctxt->om, packet.mValue, packet.mLength);
+        break;
+    }
+
+    case BLE_GATT_ACCESS_OP_WRITE_DSC:
+    {
+        // dispatch_otPlatBleGattServerOnSubscribeRequest(otInstance *aInstance, uint16_t aHandle, bool aSubscribing);
+        break;
+    }
+    }
+
+    return rc;
+}
+
+//=============================================================================
+//                        GATT SERVER
+//=============================================================================
+
+static otError bleGattServerCharacteristicRegister(otInstance *                 aInstance,
+                                                   uint16_t                     aServiceHandle,
+                                                   otPlatBleGattCharacteristic *aChar,
+                                                   bool                         aCccd)
+{
+    (void)aInstance;
+    (void)aCccd;
+
+    ble_uuid_any_t *         uuid;
+    struct ble_gatt_svc_def *svc;
+    struct ble_gatt_chr_def *chr;
+
+    if (aServiceHandle >= sNimbleServicesCount)
+        return OT_ERROR_INVALID_STATE;
+
+    svc  = &sNimbleServices[aServiceHandle];
+    chr  = &sNimbleCharacteristics[sNimbleCharacteristicsCount];
+    uuid = &sNimbleUuids[sNimbleUuidsCount];
+
+    mapOtToNimbleUuid(&aChar->mUuid, uuid);
+
+    chr->access_cb  = gatt_event_cb;
+    chr->uuid       = &uuid->u;
+    chr->flags      = aChar->mProperties;
+    chr->val_handle = &aChar->mHandleValue; // Nimble auto-fill handles in otPlatBleGattCharacteristic
+
+    if (svc->characteristics == NULL)
+    {
+        svc->characteristics = chr;
+    }
+
+    sNimbleUuidsCount++;
+    sNimbleCharacteristicsCount++;
+
+    chr           = &sNimbleCharacteristics[sNimbleCharacteristicsCount];
+    *(void **)chr = NULL;
+    // chr->uuid = 0;
+
+    // Nimble only supports one-time registration of entire GATT database.
+    return OT_ERROR_NOT_IMPLEMENTED;
```

Comment:
This function is used in otPlatBleGattServerServicesRegister. We should either return OT_ERROR_NONE or make it void function.